10up/autoshare-for-twitter
# Added Autotweet editor panel for enable/disable Autotweet and color-coded icon in Pre-Publish panel

### Description of the Change
As requested on #176, This PR adds slotfill for auto-tweet in the editor sidebar and also added a color-coded icon in the pre-publish panel. Please check screenshots for more information on this.

|Editor Sidebar ||
|---|---|
|Auto Tweet enabled |![image](https://user-images.githubusercontent.com/10613171/177468992-d00bfccb-b207-470f-b702-8656de309ddb.png)|
|Auto Tweet Disabled |![image](https://user-images.githubusercontent.com/10613171/177469119-81c6d64d-3165-4806-830b-450a1012af75.png)|
|[Status] Tweeted Successfully|![image](https://user-images.githubusercontent.com/10613171/177469706-f7b3a1ec-017e-4cc4-815e-235eca3137ce.png)|
|[Status] Tweet Failed|![image](https://user-images.githubusercontent.com/10613171/177469663-891d9351-7bae-4ad9-8a82-bfc828b7a163.png)|
|[Status] Not Tweeted|![image](https://user-images.githubusercontent.com/10613171/177469780-b1f9c38f-e11e-47dd-844f-4b2aab02977a.png)|


|Pre-Publish Panel ||
|---|---|
|Auto Tweet enabled |![image](https://user-images.githubusercontent.com/10613171/177469453-746d42c1-e8c4-4c9c-a4c7-7290669ffb71.png)|
|Auto Tweet Disabled |![image](https://user-images.githubusercontent.com/10613171/177469484-74bd428a-88da-483d-ae4a-8465e9c1e477.png)|


Closes #176 

### Alternate Designs

### Possible Drawbacks
n/a

### Verification Process
1. Go To Add a new post.
2. Notice the dedicated panel for auto-tweet in the editor sidebar.
3. Test enable/disable tweet on publish
4. Click on publish
5. Notice the pre-publish panel for auto-tweet with icon
6. Test enable/disable tweet on publish
7. Publish the post
8. Verify the tweet status in panel of the editor sidebar (for failed, success, and not tweeted).

### Checklist:






### Changelog Entry
> Added - Autotweet panel in editor sidebar for enable/disable tweet on publish.


### Credits


Props @iamdharmesh @jeffpaul @linawiezkowiak @oszkarnagy (for the UX/Design work)
